migrate/20240218_add_concession.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table(:concession) do
+      column :id, :uuid, primary_key: true, default: Sequel.lit("gen_random_uuid()")
+      column :project_id, :project, type: :uuid, null: false
+      column :resource_type, :text, collate: '"C"'
```

Comment:
I have some thoughts on this;

- Instead of resource type, I think we should give concession on billing_rate_id, which is how we track SKUs.
- This also needs to be array, so that we can group different SKUs. For example we would want to give credit for both PostgresCores and PostgresStorages combined (not separately).
- It would be good to be able to give concession to all SKUs. Something like * as a special keyword would work.
- This would be a bigger change, but maybe we should even hierarchically organize our billing rates. Then you can have * at every level of hierarchy.

Reply:
We should also define strict apply order for concessions. Because multiple rows in concession table might be applicable for given resource. For example; most specific to most general could be one way of ordering.

Reply:
Or maybe we should add a exclusion constraint to ensure there is only one concession per project per billing rate